### ISSUE TITLE
Use the width of the widget instead of progress.term_width

### ISSUE THREAD

**Issue by ritze** [WoLpH/python-progressbar#196]
#### Description

In `WidthWidgetMixin.check_size` `min_width`and `max_width` are checked against `progress.term_width`. I think this is wrong. It's more useful to check this against the current width of the widget itself.

E.g. you won't display the progressbar if there are less than 7 characters.

`term_width`: 28, `width` of progressbar = 7
```
MyVeryLongTitle |#### |  80%
```

`term_width`: 27, `width` of progressbar = 6
```
MyVeryLongTitle 80%
```

#### Code

```
class WidthWidgetMixin(object):
    '''Mixing to make sure widgets are only visible if the screen is within a
    specified size range so the progressbar fits on both large and small
    screens..
    '''

    def __init__(self, min_width=None, max_width=None, **kwargs):
        self.min_width = min_width
        self.max_width = max_width

    def check_size(self, progress):
        if self.min_width and self.min_width > progress.term_width:
            return False
        elif self.max_width and self.max_width < progress.term_width:
            return False
        else:
            return True
```

**Comment by WoLpH** [comment]
That would make a bit more sense indeed. I'm not entirely sure how easy it is to implement though, it might give inaccurate results.

The problem is that if you have multiple widgets with variable size, the entire size would need to be recalculated as soon as one widget decides that it won't render. Perhaps a non-issue, not sure... but it could be a problem.

**Comment by ritze** [comment]
I'm playing a little bit around with the code and thinking about containers. So you can add multiple widgets to one container, this container is handled like a widget. Than you can set a fixed width or maybe later a relative width in percent of the terminal width for each widget and container.

There are some questions open e.g. what to do if two widgets/containers have both a relative width of 60% of the terminal width? So you're right, there would be much more work instead of just adding an additional parameters. This would also affects the way how the width of the widgets are calculated.

**Comment by WoLpH** [comment]
Implemented through #203

**Comment by WoLpH** [comment]
Implemented through #203
